src/adapter/src/coord/message_handler.rs

```diff
@@ -465,9 +465,10 @@ impl<S: Append + 'static> Coordinator<S> {
             Err(e) => {
                 // Drop the placeholder sink if still present.
                 if self.catalog.try_get_entry(&id).is_some() {
+                    let ops = self.catalog.drop_items_ops(&[id]);
```

Comment:
Its important to note that the sink _uses_ an underlying connection, whereas cascading here drops anything that are _used_by_ the sink....which is nothing because it does not yet exist; This allows us to drop the sink and the cluster;

However, if we crash/restart before we actually run this code, i think we will still leak some objects and/or a cluster; definitely a followup here, but there is probably no way to cleanup transactionally?